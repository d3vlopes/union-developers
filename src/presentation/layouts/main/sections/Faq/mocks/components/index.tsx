import { vitest } from 'vitest'

vitest.mock('@/presentation/components/organisms', () => ({
  QuestionsAccordion: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return (
      <div data-testid="mock-questions-accordion">{children}</div>
    )
  },
}))
