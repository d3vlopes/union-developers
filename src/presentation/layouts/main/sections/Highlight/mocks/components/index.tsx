import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Highlights: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-highlights">{children}</div>
  },
}))
