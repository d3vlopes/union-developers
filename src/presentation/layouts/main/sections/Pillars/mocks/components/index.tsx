import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  MethodologyCards: function Mock({
    children,
  }: {
    children: React.ReactNode
  }) {
    return <div data-testid="mock-methodology-cards">{children}</div>
  },
}))
