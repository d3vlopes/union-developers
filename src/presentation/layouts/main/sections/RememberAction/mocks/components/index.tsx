import { vitest } from 'vitest'

vitest.mock(
  '@/presentation/components/molecules/RememberAction',
  () => ({
    RememberAction: function Mock({
      children,
    }: {
      children: React.ReactNode
    }) {
      return <div data-testid="mock-remember-action">{children}</div>
    },
  }),
)
