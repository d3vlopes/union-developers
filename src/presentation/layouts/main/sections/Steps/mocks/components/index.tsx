import { vitest } from 'vitest'

vitest.mock('@/presentation/components/molecules', () => ({
  Steps: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="mock-steps">{children}</div>
  },
}))
