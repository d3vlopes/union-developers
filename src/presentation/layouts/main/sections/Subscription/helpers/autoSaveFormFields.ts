import { Dispatch, SetStateAction } from 'react'

export function autoSaveFormFields<TFormValues, TFormData>(
  formValues: TFormValues,
  formData: TFormData,
  errors: object,
  setFormData: Dispatch<SetStateAction<TFormData>>,
  delay = 5000,
) {
  const interval = setInterval(() => {
    const isFieldContainError = Object.keys(errors).length > 0

    if (!isFieldContainError) {
      setFormData({ ...formData, ...formValues })
    }
  }, delay)

  return interval
}
