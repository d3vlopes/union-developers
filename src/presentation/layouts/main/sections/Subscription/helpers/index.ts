export * from './autoSaveFormFields'
export * from './formStepMap'
