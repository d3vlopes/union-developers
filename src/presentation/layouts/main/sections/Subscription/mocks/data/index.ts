import { generateId } from '@/utils'

import { termsMock } from '../../steps/AboutYou/components/Terms/mock'

import { SubscriptionSectionProps } from '../..'

export const subscriptionMock: SubscriptionSectionProps = {
  id: 'formulario',
  title: 'Junte-se ao time!',
  initialStepDescription:
    'Tem interesse em participar do Union? Confira abaixo os requisitos para participar',
  formStepsDescription:
    'Ficamos felizes em saber que você tem interesse em fazer parte do Union! Agora é só responder o nosso formulário',
  requirements: [
    'Ter mais que 18 anos',
    'Ter 1 ano ou mais de estudo em programação',
    'Ainda não trabalhar na área',
    'Já ter criado projetos sozinho que não sejam de cursos ou eventos',
    'Não participar de bootcamps ou programas de capacitação',
    'Ter disponibilidade para trabalhar no mínimo 1 hora por dia no projeto',
    'Ter disponibilidade para participar das reuniões do time, durante a semana e finais de semana',
  ],
  buttonText: 'Inscrever-se',
  formStepNavLinks: [
    {
      id: generateId(),
      stepTitle: 'Dados pessoais',
      stepNumber: 1,
    },
    {
      id: generateId(),
      stepTitle: 'Informações técnicas',
      stepNumber: 2,
    },
    {
      id: generateId(),
      stepTitle: 'Sobre você',
      stepNumber: 3,
    },
    {
      id: generateId(),
      stepTitle: 'Enviar',
      stepNumber: 4,
    },
  ],
  terms: termsMock,
}
