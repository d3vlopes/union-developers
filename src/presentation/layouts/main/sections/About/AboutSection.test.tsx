import { render, screen } from '@/__tests__/helpers'

import { mainLayoutMock } from '@/presentation/layouts/main/mocks/data'

import { AboutSection } from '.'

const { getByRole, getByText } = screen

const mock = mainLayoutMock.aboutSection

describe('<AboutSection />', () => {
  it('should contain correct section id', () => {
    const { container } = render(<AboutSection {...mock} />)

    const wrapper = container.firstChild

    expect(wrapper).toHaveAttribute('id', mock.id)
  })

  it('should render heading', () => {
    render(<AboutSection {...mock} />)

    const heading = getByRole('heading', { name: mock.heading })

    expect(heading).toBeInTheDocument()
  })

  it('should render description', () => {
    render(<AboutSection {...mock} />)

    const description = getByText(mock.description)

    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<AboutSection {...mock} />)

    expect(container).toMatchSnapshot()
  })
})
