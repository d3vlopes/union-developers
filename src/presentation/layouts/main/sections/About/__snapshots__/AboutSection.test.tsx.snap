// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AboutSection /> > should match snapshot 1`] = `
.c0 {
  position: relative;
  margin-top: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #1F1D2B;
  background-clip: padding-box;
  border: 3px solid transparent;
  border-radius: 20px;
  width: 100%;
  -webkit-scroll-margin-block-start: 150px;
  -moz-scroll-margin-block-start: 150px;
  -ms-scroll-margin-block-start: 150px;
  scroll-margin-block-start: 150px;
  -webkit-scroll-margin-block-end: 150px;
  -moz-scroll-margin-block-end: 150px;
  -ms-scroll-margin-block-end: 150px;
  scroll-margin-block-end: 150px;
}

.c0:before {
  content: '';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -1;
  margin: -3px;
  border-radius: inherit;
  background: linear-gradient(-74deg,#0048FE -50%,#851D86 100%);
}

.c1 {
  position: absolute;
  display: inline-block;
  background: #1F1D2B;
  margin: -18px 0 0 22px;
  padding: 0 5px 0 8px;
}

.c1 h2 {
  font-size: 2rem;
  font-weight: 600;
  background: linear-gradient(225deg,#0048FE -50%,#851D86 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c2 {
  padding: 3.5rem 1.6rem 2.4rem;
}

.c2 p {
  color: #E2E2E2;
  text-align: center;
  line-height: 150%;
}

.c2 p:not(:last-child) {
  padding-bottom: 2.4rem;
}

.c3 {
  position: absolute;
  bottom: -55px;
  left: 50%;
  border-radius: 0 0 20px 20px;
  background: linear-gradient(150deg,#851D86 -115%,#0048FE 280%);
  width: 5px;
  height: 64px;
  z-index: -2;
}

.c3:after {
  content: '';
  position: absolute;
  width: inherit;
  height: inherit;
  border-radius: inherit;
  background: linear-gradient( 320deg, #1F1D2B 0%, rgba(31,29,43,0) 100% );
}

@media (min-width:768px) {
  .c0 {
    max-width: 125.6rem;
    margin-top: calc(3.2rem + 5rem);
  }

  .c0:before {
    background: linear-gradient(-74deg,#0048FE -10%,#851D86 100%);
  }
}

@media (min-width:768px) {
  .c1 {
    margin: -35px 0 0 85px;
  }

  .c1 h2 {
    font-size: 4rem;
  }
}

@media (min-width:768px) {
  .c2 {
    padding: 5rem 8rem 5rem;
  }

  .c2 p {
    font-size: 2.4rem;
    max-width: 102.7rem;
    text-align: left;
  }
}

@media (min-width:768px) {
  .c3 {
    bottom: -88px;
    height: 88px;
    background: linear-gradient(0deg,#851D86 -70%,#0048FE 280%);
  }
}

<div>
  <section
    class="c0"
    id="about"
  >
    <header
      class="c1"
    >
      <h2>
        Quem somos?
      </h2>
    </header>
    <div
      class="c2"
    >
      O Union é um grupo que tem como missão proporcionar experiências para desenvolvedores em inicio de carreira, oferecendo um ambiente prático e imersivo próximo do encontrado nas empresas, desafiando seus membros a resolver problemas reais, com cronogramas e prazos de entrega. No Union, você desenvolve as skills mais procuradas pelo mercado, trabalha com ferramentas e metodologias utilizada pelas empresas, além de ganhar experiência trabalhando com outras pessoas de varias partes do Brasil. Aqui você vai vivenciar na prática como é o dia a dia de um desenvolver em uma empresa e vai ficar mais preparado para conquistar sua vaga.
    </div>
    <div
      class="c3"
    />
  </section>
</div>
`;
